hoss/coordinate_utilities.py

```diff
@@ -101,20 +101,23 @@ def get_coordinate_variables(
     CF-Convention coordinates metadata attribute. It returns them in a specific
     order [latitude, longitude]"
     """
-
-    coordinate_variables_list = varinfo.get_references_for_attribute(
+    # varinfo returns a set and not a list
+    coordinate_variables = varinfo.get_references_for_attribute(
         requested_variables, 'coordinates'
     )
+
     latitude_coordinate_variables = [
         coordinate
-        for coordinate in coordinate_variables_list
-        if varinfo.get_variable(coordinate).is_latitude()
+        for coordinate in coordinate_variables
+        if varinfo.get_variable(coordinate) is not None
```

Comment:
Separate comment - you have not added unit tests to prove this case is being handled as expected. (Needed both for latitude or longitude)

Reply:
- this method is first called before prefetch. it should not fail if the variables don't exist. 
- if it is present in the variable and not in the file - it will fail during prefetch. That is when it is first accessed from the file. - will check if I have a test on that
- There is a unit test for the function - will add subtests if the variables don't exist

Reply:
There are unit tests to check for the above cases in line 205 of test_coordinate_utilities.py

Reply:
ok. but the failure case is in the function that calls it, not  get_coordinate_variables itself right? So the test has to be  in the functions that call it

Reply:
added additional unit tests - https://github.com/nasa/harmony-opendap-subsetter/commit/913004d85de47c3197855398d03e85a52d27f160

Reply:
Nice - thanks!